### PR TITLE
feat(cli): build before migrate and openapi-spec scripts

### PR DESCRIPTION
**Pull request by hacksparrow**
`npm run migrate` and `openapi-spec` access the `./dist/migrate` directory, which won't be available if the app has never been built or started.

Users should not be required to start the app once or type `npm run build` before they can use `npm run migrate`. Let's build the app before every migration or openapi spec generation command. Let's make `npm run migrate` and `openapi-spec` just work.

Building the app before every migration or spec generation command will also result in using the latest changes in the source files. Not doing so (currently) causes the commands to use oudated files from `./dist/migrate`, this gives the wrong impression that the migration process is not working as expected.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: examples/access-control-migration/package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "test": "lb-mocha \"dist/__tests__/**/*.js\"",
     "test:dev": "lb-mocha --allow-console-logs dist/__tests__/**/*.js",
     "verify": "npm pack && tar xf loopback-access-control-migration*.tgz && tree package && npm run clean",
-    "migrate": "node ./dist/migrate",
-    "openapi-spec": "node ./dist/openapi-spec",
+    "migrate": "npm run build && node ./dist/migrate",
```
Comment (bajtos) [inline examples/access-control-migration/package.json:31]: Have you considered using a `premigrate` script instead?

```suggestion
    "premigrate": "npm run build",
    "migrate": "node ./dist/migrate",
```

See how we are using `pretest` and `prestart` to run `npm run build` before the tests, and [Pre & Post Scripts](https://docs.npmjs.com/using-npm/scripts#pre--post-scripts) in npmjs docs.

(Same comment applies to `openapi-spec`, I think the script would be called `preopenapi-spec`.)

**Review by bajtos — COMMENTED** [review]
Great improvement 👏🏻

**Review by jannyHou — APPROVED** [review]
LGTM and please consider Miroslav's suggestion about the pre script.

File: examples/access-control-migration/package.json (hunk 1 above)
Comment (hacksparrow) [reply examples/access-control-migration/package.json]: @bajtos thanks, done.

File: examples/access-control-migration/package.json [hunk 2]
```diff
@@ -29,7 +29,9 @@
     "test:dev": "lb-mocha --allow-console-logs dist/__tests__/**/*.js",
     "verify": "npm pack && tar xf loopback-access-control-migration*.tgz && tree package && npm run clean",
     "migrate": "node ./dist/migrate",
+    "premigrate": "npm run build ",
     "openapi-spec": "node ./dist/openapi-spec",
+    "preopenapi-spec": "npm run build",
```
Comment (raymondfeng) [inline examples/access-control-migration/package.json:34]: Nitpick: 

- move `premigrate` to come before `migrate`
- move `preopenapi-spec` to come before `openapi-spec`

**Comment by dhmlau** [comment]
Related to https://github.com/strongloop/loopback-next/issues/4189.
Actually we also need to run the build before running `lb4 discover` as well. Do you think you can fix it as well? Perhaps a separate PR since it's out of scope for this PR? Thanks.

File: examples/access-control-migration/package.json (hunk 1 above)
Comment (hacksparrow) [reply examples/access-control-migration/package.json]: @bajtos looks good now?

**Comment by hacksparrow** [comment]
> Actually we also need to run the build before running lb4 discover as well. Do you think you can fix it as well? Perhaps a separate PR since it's out of scope for this PR? Thanks.

The implementation is more involved, will do in a separate PR.

File: examples/access-control-migration/package.json (hunk 2 above)
Comment (hacksparrow) [reply examples/access-control-migration/package.json:33]: Done.

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**